### DISCUSSION TITLE
Endpoint for ILM (in data streams)

### DISCUSSION THREAD

**Discussion by evgeniy-r** [fluent/fluentd#4053] [Q&A]
Hello!

The output plugin `elasticsearch_data_stream` uses the `_enrich/policy` endpoint for newer versions of the gem ([link](https://github.com/uken/fluent-plugin-elasticsearch/blob/319b6f8cf102b5fbca83e00e37d1e0bc2444c8cb/lib/fluent/plugin/out_elasticsearch_data_stream.rb#L106)):
```ruby
 client(host).enrich.put_policy(params.merge(name: ilm_name))
```
But it seems this endpoint is for [enrich policies](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-enrich-policy-api.html), not ILM policies.

For older versions it uses  the `_ilm/policy` endpoint.
```ruby
client(host).xpack.ilm.put_policy(params.merge(policy_id: ilm_name))
```

When I try to create ILM via Fluent config I get errors like this: `[400] {"error":{"root_cause":[{"type":"x_content_parse_exception","reason":"[1:15] [policy] unknown field [phases]"}],"type":"x_content_parse_exception",...`
And it's correct - there is no such field for enrich policies.

I'm a bit confused, because enrich policies and index lifecycle management policies (ILM policies) are different things.

P. S. I know that it's about the other project, but may be somebody still can answer.

**Comment by norelm** [comment]
Hello, i have this exact issue, did you get a solution?

**Reply by evgeniy-r** [reply]
Hi, no (I just have created ILM through Kibana).
